status-im/status-react
# [#17896] Raw balances using big number

fixes #17896

### Summary

The previous implementation was using js/parseInt to process the raw-balances, this PR stores them as big-number instances in re-frame.

This had some problems, if we had 100 or more units of a token, the calculations resulted in a `0`, now using bignumbers that issue has been fixed.

### Changes
- Makes `utils.money/add` able to receive `nil` values
- Store raw-balances as big number instances
- Update the balance calculation related functions to use big numbers, also now they are using existing `utils.money` functions.

### Testing notes
 
 The app should work as expected. Previously, if we had 100 or more token units, the result listed in the account cards in the wallet home page was `$0.00`. Now it shows the proper value.

#### Platforms

- Android
- iOS

#### Areas that maybe impacted

Wallet home page in the account cards.

### Steps to test


- Open Status
- Navigate to our new wallet implementation
- Wait for it to load, the account cards should properly display the account balance if we have more than 100 token units in total for an account.

status: ready
